kratos/solving_strategies/builder_and_solvers/builder_and_solver.h

```diff
@@ -164,31 +164,31 @@ class BuilderAndSolver
+      bool Are_RHS_Element_Components_Set() const { if( mpRHS_Element_Variables == NULL ) return false; else return true; };
 
       // Conditions
       void SetLHS_Condition_Components ( std::vector<TSystemMatrixType>& rLHS_Condition_Components ) { mpLHS_Condition_Components = &rLHS_Condition_Components; };
       void SetLHS_Condition_Variables     ( const std::vector< Variable< LocalSystemMatrixType > >& rLHS_Condition_Variables ) { mpLHS_Condition_Variables = &rLHS_Condition_Variables; };
       void SetRHS_Condition_Components ( std::vector<TSystemVectorType>& rRHS_Condition_Components ) { mpRHS_Condition_Components = &rRHS_Condition_Components; };
       void SetRHS_Condition_Variables     ( const std::vector< Variable< LocalSystemVectorType > >& rRHS_Condition_Variables ) { mpRHS_Condition_Variables = &rRHS_Condition_Variables; };
 
-      bool Are_LHS_Condition_Components_Set() { if( mpLHS_Condition_Variables == NULL ) return false; else return true; };
-      bool Are_RHS_Condition_Components_Set() { if( mpRHS_Condition_Variables == NULL ) return false; else return true; };
+      bool Are_LHS_Condition_Components_Set() const { if( mpLHS_Condition_Variables == NULL ) return false; else return true; };
+      bool Are_RHS_Condition_Components_Set() const { if( mpRHS_Condition_Variables == NULL ) return false; else return true; };
 
       //getting pointer variables
 
       // Elements
       std::vector<TSystemMatrixType>& GetLHS_Element_Components() { return *mpLHS_Element_Components; };
-      const std::vector< Variable< LocalSystemMatrixType > >& GetLHS_Element_Variables() { return *mpLHS_Element_Variables; };
+      const std::vector< Variable< LocalSystemMatrixType > >& GetLHS_Element_Variables() const { return *mpLHS_Element_Variables; };
       std::vector<TSystemVectorType>& GetRHS_Element_Components() { return *mpRHS_Element_Components; };
-      const std::vector< Variable< LocalSystemVectorType > >& GetRHS_Element_Variables() { return *mpRHS_Element_Variables; };
+      const std::vector< Variable< LocalSystemVectorType > >& GetRHS_Element_Variables() const { return *mpRHS_Element_Variables; };
 
       // Conditions
       std::vector<TSystemMatrixType>& GetLHS_Condition_Components() { return *mpLHS_Condition_Components; };
-      const std::vector< Variable< LocalSystemMatrixType > >& GetLHS_Condition_Variables() { return *mpLHS_Condition_Variables; };
+      const std::vector< Variable< LocalSystemMatrixType > >& GetLHS_Condition_Variables() const { return *mpLHS_Condition_Variables; };
       std::vector<TSystemVectorType>& GetRHS_Condition_Components() { return *mpRHS_Condition_Components; };
-      const std::vector< Variable< LocalSystemVectorType > >& GetRHS_Condition_Variables() { return *mpRHS_Condition_Variables; };
+      const std::vector< Variable< LocalSystemVectorType > >& GetRHS_Condition_Variables() const { return *mpRHS_Condition_Variables; };
```

Comment:
puh those are super old, I was actually hoping to remove those at some point since they don't seem to be used any more